tests/Console/Command/AddPrefixCommandTest.php

```diff
@@ -0,0 +1,254 @@
+        );
+    }
+
+    public function test_can_scope_a_file()
+    {
+        $this->appTester->run(
+            [
+                'add-prefix',
+                'prefix' => 'MyPrefix\\\\',
+                'paths' => ['dir'.DIRECTORY_SEPARATOR.'dir'.DIRECTORY_SEPARATOR.'MyClass.php'],
+            ],
+            ['capture_stderr_separately' => true]
+        );
+
+        $expected = <<<EOF
+Scoping $this->tempDir/dir/dir/MyClass.php. . . Success
+
+EOF;
+        $expected = str_replace('/', DIRECTORY_SEPARATOR, $expected);
+
+        Assert::assertSame(0, $this->appTester->getStatusCode());
+        Assert::assertStringEndsWith($expected, $this->appTester->getDisplay(true));
+        Assert::assertEmpty($this->appTester->getErrorOutput(true));
+
+        Assert::assertFileEquals(
+            self::FIXTURES_DIR.'/replaced/dir/dir/MyClass.php',
+            $this->tempDir.'/dir/dir/MyClass.php'
+        );
+    }
+
+    public function test_can_scope_for_multiple_paths()
+    {
+        $this->appTester->run(
+            [
+                'add-prefix',
+                'prefix' => 'MyPrefix\\\\',
+                'paths' => [
+                    'dir'.DIRECTORY_SEPARATOR.'dir'.DIRECTORY_SEPARATOR.'MyClass.php',
+                    'dir'.DIRECTORY_SEPARATOR.'dir'.DIRECTORY_SEPARATOR.'MySecondClass.php',
+                ],
+            ],
+            ['capture_stderr_separately' => true]
+        );
+
+        $expected = <<<EOF
+Scoping $this->tempDir/dir/dir/MyClass.php. . . Success
+Scoping $this->tempDir/dir/dir/MySecondClass.php. . . Success
+
+EOF;
+        $expected = str_replace('/', DIRECTORY_SEPARATOR, $expected);
+
+        Assert::assertSame(0, $this->appTester->getStatusCode());
+        Assert::assertStringEndsWith($expected, $this->appTester->getDisplay(true));
+        Assert::assertEmpty($this->appTester->getErrorOutput(true));
+
+        Assert::assertFileEquals(
+            self::FIXTURES_DIR.'/replaced/dir/dir/MyClass.php',
+            $this->tempDir.'/dir/dir/MyClass.php'
+        );
+
+        Assert::assertFileEquals(
+            self::FIXTURES_DIR.'/replaced/dir/dir/MySecondClass.php',
+            $this->tempDir.'/dir/dir/MySecondClass.php'
+        );
+    }
+
+    public function test_ignore_invalid_files()
+    {
+        $this->appTester->run(
+            [
+                'add-prefix',
+                'prefix' => 'MyPrefix\\\\',
+                'paths' => ['dir'.DIRECTORY_SEPARATOR.'MyIncorrectClass.php'],
+            ],
+            ['capture_stderr_separately' => true]
+        );
+
+        $expected = <<<EOF
+Scoping $this->tempDir/dir/MyIncorrectClass.php. . . Fail
+
+EOF;
+        $expected = str_replace('/', DIRECTORY_SEPARATOR, $expected);
+
+        Assert::assertSame(0, $this->appTester->getStatusCode());
+        Assert::assertStringEndsWith($expected, $this->appTester->getDisplay(true));
+    }
+
+    public function test_cannot_scope_nonexistent_files()
```

Comment:
There should be a try/catch in the code, it's easier to test the status code and previous exception